internal/cache/async.go

```diff
@@ -149,3 +156,31 @@ func isNamespaceTerminating(err error) bool {
 	return (errors.IsForbidden(err) && namespaceTerminatingPattern.FindString(err.Error()) != "") ||
 		(errors.IsNotFound(err) && namespaceNotFoundPattern.FindString(err.Error()) != "")
 }
+
+const (
+	asyncClientRequest				= "foundry.spark.scheduler.async.request.count"
+	asyncClientRetries				= "foundry.spark.scheduler.async.request.retries.count"
+	asyncClientDroppedRequests		= "foundry.spark.scheduler.async.request.dropped.count"
+	objectTypeTagKey                = "objectType"
+)
+
+// AsyncClientMetrics emits metrics on retries and failures of the internal async client calls to the api server
+// TODO: Move this to the metrics package after a refactor of that package to avoid cyclical imports
+type AsyncClientMetrics struct {
+	ObjectTypeTag string
+}
+
+// MarkRequest marks that a request to the api server is being made
+func (acm *AsyncClientMetrics) MarkRequest(ctx context.Context) {
```

Comment:
I think we should also tag with the request type

Reply:
Done